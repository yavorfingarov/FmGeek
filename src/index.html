<!doctype html>
<html lang="en">
    <head>
        <meta charset="UTF-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <meta name="description" content="Listen to your favorite radio stations!" />
        <meta name="keywords" content="radio, music, app, pwa" />
        <meta name="author" content="Yavor Fingarov" />
        <meta name="theme-color" content="#0d47a1" />
        <title x-data x-text="$store.player.current?.name ?? 'FmGeek'"></title>
        <link rel="icon" href="./icons/favicon.ico" sizes="48x48" />
        <link rel="icon" href="./icons/favicon.svg" sizes="any" type="image/svg+xml" />
        <link rel="apple-touch-icon" href="./icons/apple-touch-icon-180x180.png" />
        <link rel="stylesheet" href="./style.scss" />
        <script type="module" src="./main.js"></script>
    </head>
    <body x-data :class="$store.settings.ui.theme">
        <noscript>
            <p class="center error">Please enable JavaScript to use this application.</p>
        </noscript>
        <main x-data x-cloak>
            <!---->
            {{> header }}
            <!---->
            {{> navigation }}
            <!---->
            {{> settings }}
            <!---->
            {{> about }}
            <!---->
            {{> privacy }}
            <!---->
            {{> player }}
            <!---->
        </main>
    </body>
</html>
