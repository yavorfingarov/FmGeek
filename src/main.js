import Alpine from "alpinejs";
import persist from "@alpinejs/persist";
import {header} from "./components/header";
import {navigation} from "./components/navigation";
import {settings} from "./components/settings";
import {player} from "./components/player";

window.Alpine = Alpine;
Alpine.plugin(persist);
header();
navigation();
settings();
player();
Alpine.start();
