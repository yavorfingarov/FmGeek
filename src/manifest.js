export const manifest = {
    name: "FmGeek",
    short_name: "FmGeek",
    description: "Listen to your favorite radio stations!",
    start_url: "/",
    display: "standalone",
    theme_color: "#0d47a1",
    background_color: "#0d47a1",
    icons: [
        {
            src: "pwa-64x64.png",
            sizes: "64x64",
            type: "image/png"
        },
        {
            src: "pwa-192x192.png",
            sizes: "192x192",
            type: "image/png"
        },
        {
            src: "pwa-512x512.png",
            sizes: "512x512",
            type: "image/png"
        },
        {
            src: "maskable-icon-512x512.png",
            sizes: "512x512",
            type: "image/png",
            purpose: "maskable"
        }
    ]
};
