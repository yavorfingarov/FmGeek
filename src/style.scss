@import url("../node_modules/simpledotcss/simple.css");

html.light,
html.auto {
    color-scheme: light;
    --bg: #fefefe;
    --accent-bg: #f5f7ff;
    --text: #212121;
    --text-light: #585858;
    --border: #898ea4;
    --accent: #0d47a1;
    --accent-hover: #1266e2;
    --accent-text: var(--bg);
    --code: #d81b60;
    --preformatted: #444;
    --marked: #ffdd33;
    --disabled: #efefef;
    img {
        filter: opacity(75%);
    }
}

@mixin dark {
    color-scheme: dark;
    --bg: #212121;
    --accent-bg: #2b2b2b;
    --text: #dcdcdc;
    --text-light: #ababab;
    --accent: #ffb300;
    --accent-hover: #ffe099;
    --accent-text: var(--bg);
    --code: #f06292;
    --preformatted: #ccc;
    --disabled: #111;
    img {
        filter: invert(75%);
    }
}

html.dark {
    @include dark();
}

@media (prefers-color-scheme: dark) {
    html.auto {
        @include dark();
    }
}

html {
    scrollbar-gutter: stable;
}

body {
    -webkit-tap-highlight-color: transparent;
    user-select: none;
}

main {
    padding-bottom: 1.5rem;
}

section {
    padding: 0;
    border: 0;
    margin: 0;
}

img {
    display: block;
    &.smaller {
        height: 32px;
        width: 32px;
    }
}

h1,
h2 {
    font-weight: 700;
    line-height: normal;
}

h1 {
    font-size: 1.5em;
}

h2 {
    font-size: 1em;
}

input,
select {
    width: 10em;
}

button.flat {
    color: var(--accent);
    background-color: inherit;
    border: none;
    &:hover {
        color: var(--accent-hover);
        background-color: inherit;
    }
    &.no-hover-effect:hover {
        color: var(--accent);
    }
}

textarea {
    font-family: var(--mono-font);
    height: 10em;
    overflow: auto;
    white-space: pre;
    resize: vertical;
}

.flex {
    display: flex;

    &.column {
        flex-direction: column;
    }

    &.space-between {
        justify-content: space-between;
    }

    &.center {
        justify-content: center;
    }

    &.space-evenly {
        justify-content: space-evenly;
    }

    &.center-align {
        align-items: center;
    }

    &.wrap {
        flex-wrap: wrap;
    }

    &.gap {
        gap: 1.5em;
    }
}

.padding {
    padding: 0.3em;
}

.no-padding {
    padding: 0 !important;
}

.margin {
    margin: 1.5em 0;
}

.no-margin {
    margin: 0 !important;
}

.center {
    text-align: center;
}

.smaller {
    font-size: 0.9em;
}

.error {
    color: var(--code);
}

.no-overflow {
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;
}

.no-decoration {
    text-decoration: none;
}

[x-cloak] {
    display: none !important;
}
