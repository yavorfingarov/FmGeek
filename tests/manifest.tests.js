import {expect, test} from "vitest";
import {manifest} from "../src/manifest";

test("snapshot", function () {
    expect(manifest).toMatchSnapshot();
});
