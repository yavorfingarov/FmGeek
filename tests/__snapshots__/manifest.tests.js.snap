// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
{
  "background_color": "#0d47a1",
  "description": "Listen to your favorite radio stations!",
  "display": "standalone",
  "icons": [
    {
      "sizes": "64x64",
      "src": "pwa-64x64.png",
      "type": "image/png",
    },
    {
      "sizes": "192x192",
      "src": "pwa-192x192.png",
      "type": "image/png",
    },
    {
      "purpose": "any",
      "sizes": "512x512",
      "src": "pwa-512x512.png",
      "type": "image/png",
    },
    {
      "purpose": "maskable",
      "sizes": "512x512",
      "src": "maskable-icon-512x512.png",
      "type": "image/png",
    },
  ],
  "name": "FmGeek",
  "short_name": "FmGeek",
  "start_url": "/",
  "theme_color": "#0d47a1",
}
`;
